tests/components/hue/test_sensor_base.py

```diff
@@ -466,3 +501,121 @@ def mock_bridge(hass):
     assert len(mock_bridge.mock_requests) == 0
     assert len(hass.states.async_all()) == 0
     assert len(mock_bridge.handle_unauthorized_error.mock_calls) == 1
+
+
+async def test_hue_events(hass, mock_bridge):
+    """Test that hue remotes fire events when pressed."""
+    mock_bridge.mock_sensor_responses.append(SENSOR_RESPONSE)
+
+    mock_listener = Mock()
+    unsub = hass.bus.async_listen(CONF_HUE_EVENT, mock_listener)
+
+    await setup_bridge(hass, mock_bridge)
+    assert len(mock_bridge.mock_requests) == 1
+    assert len(hass.states.async_all()) == 7
+    assert len(mock_listener.mock_calls) == 0
+
+    new_sensor_response = dict(SENSOR_RESPONSE)
+    new_sensor_response["7"]["state"] = {
+        "buttonevent": 18,
+        "lastupdated": "2019-12-28T22:58:02",
+    }
+    mock_bridge.mock_sensor_responses.append(new_sensor_response)
+
+    # Force updates to run again
+    await mock_bridge.sensor_manager.coordinator.async_refresh()
```

Comment:
It's better to patch time and fire a time changed event to have the coordinator do a refresh. Then we're not relying on the details in our implementation in the tests.

Reply:
I just mirrored the way it is done in the other unit tests in this same file, where forced coordinator calls are being made like that :(

Reply:
Yeah, my comment is for the future.